lib/assets/javascripts/turbograft/response.coffee

```diff
@@ -0,0 +1,26 @@
+class TurboGraft.Response
+  constructor: (@xhr) ->
+
+  valid: -> !@clientOrServerError() && @validContent()
+
+  document: ->
+    if @valid()
+      TurboGraft.Document.create(@xhr.responseText)
+
+  clientOrServerError: ->
```

Comment:
what if we call this `hasRenderableHttpStatus`?

`@xhr.status == 422 || (200 <= @xhr.status < 400)`

mainly for philosophical reasons (422 is a client error lol)